src/main/java/oracle/kubernetes/operator/builders/WatchBuilder.java

```diff
@@ -0,0 +1,320 @@
+        return FACTORY.createWatch(clientHolder, callParams, Domain.class, WatchBuilder::listDomainsForAllNamespacesCall);
+    }
+
+    private static Call listDomainsForAllNamespacesCall(ClientHolder clientHolder, CallParams callParams) {
+        try {
+            return clientHolder.getWeblogicApiClient().listWebLogicOracleV1DomainForAllNamespacesCall(START_LIST,
+                        callParams.getFieldSelector(), callParams.getIncludeUninitialized(), callParams.getLabelSelector(),
+                        callParams.getLimit(), callParams.getPretty(), callParams.getResourceVersion(),
+                        callParams.getTimeoutSeconds(), WATCH, callParams.getProgressListener(), callParams.getProgressRequestListener());
+        } catch (ApiException e) {
+            throw new UncheckedApiException(e);
+        }
+    }
+
+    /**
+     * Creates a web hook object to track changes to weblogic domains in one namespaces
+     * @param namespace the namespace in which to track domains
+     * @return the active web hook
+     * @throws ApiException if there is an error on the call that sets up the web hook.
+     */
+    public Watch<Domain> createDomainsInNamespaceWatch(String namespace) throws ApiException {
+        return FACTORY.createWatch(clientHolder, callParams, Domain.class, new ListDomainsInNamespaceCall(namespace));
+    }
+
+    private class ListDomainsInNamespaceCall implements BiFunction<ClientHolder, CallParams, Call> {
+        private String namespace;
+
+        ListDomainsInNamespaceCall(String namespace) {
+            this.namespace = namespace;
+        }
+
+        @Override
+        public Call apply(ClientHolder clientHolder, CallParams callParams) {
+            try {
+                return clientHolder.getWeblogicApiClient().listWebLogicOracleV1NamespacedDomainCall(namespace,
+                            callParams.getPretty(), START_LIST, callParams.getFieldSelector(),
+                            callParams.getIncludeUninitialized(), callParams.getLabelSelector(), callParams.getLimit(),
+                            callParams.getResourceVersion(), callParams.getTimeoutSeconds(), WATCH, null, null);
+            } catch (ApiException e) {
+                throw new UncheckedApiException(e);
+            }
+        }
+    }
+
+    /**
+     * Creates a web hook object to track changes to custom objects
+     * @param namespace the namespace in which to track custom objects.
+     * @param responseBodyType the type of objects returned in the events
+     * @param group the custom resource group name
+     * @param version the custom resource version
+     * @param plural the custom resource plural name    @return the active web hook
+     * @throws ApiException if there is an error on the call that sets up the web hook.
+     */
+    public <T> Watch<T> createCustomObjectWatch(String namespace, Class<?> responseBodyType, String group, String version, String plural) throws ApiException {
```

Comment:
Similar...  I think the test that uses this is Bill's very first test.  We only have one "custom" type, which is Domain.  We would never watch or list for generic custom types.